Assets/Scripts/Extra/unityChanIdleHandler_shootingRange.cs

```diff
@@ -0,0 +1,47 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class unityChanIdleHandler_shootingRange : MonoBehaviour
+{
+    [SerializeField] private Vector2Int idleRange = Vector2Int.zero;
+    [SerializeField] private Vector2 changeIdleAnimationRange = Vector2.zero;
+    private Animator currentAnimator = null;
+    private bool mPA_isStarting = false;
+
+    private void Start()
+    {
+        currentAnimator = GetComponent<Animator>();
+        StartCoroutine(ChangeIdleAnimationCour());
+    }
+
+    public void Start_MPA()
+    {
+        mPA_isStarting = true;
+        ChangeIdleAnimation(0);
+    }
+
+    public void End_MPA()
+    {
+        mPA_isStarting = false;
+    }
+
+    private IEnumerator ChangeIdleAnimationCour()
+    {
+        while (true)
+        {
+            yield return new WaitForSeconds(Random.Range(changeIdleAnimationRange.x, changeIdleAnimationRange.y));
```

Comment:
This is a little problematic. It will make the script count the time even while idle variant is currently being played. Usually you'd want the `changeIdleAnimationRange` to be the wait time _in-between_ playing those idle variants. Meaning, you count the time only while the Animator is in the WAIT00 state.

This is what I'd do instead:
```suggestion
            float waitTime = Random.Range(changeIdleAnimationRange.x, changeIdleAnimationRange.y);

            while (waitTime > 0)
            {
                // deduct wait time only while in idle state
                if (currentAnimator.GetCurrentAnimatorStateInfo(0).IsName("WAIT00")) // you may want to replace "WAIT00" with a string variable instead
                {
                    waitTime -= Time.deltaTime;
                }

                yield return null;
            }
```